packages/entity/src/EntityMutator.ts

```diff
@@ -72,6 +74,19 @@ abstract class BaseMutator<
     protected readonly metricsAdapter: IEntityMetricsAdapter
   ) {}
 
+  protected validateFields(fields: Partial<TFields>): Result<void> {
+    for (const fieldName in fields) {
+      const fieldValue = fields[fieldName];
+      const fieldDefinition = this.entityConfiguration.schema.get(fieldName);
+      invariant(fieldDefinition, `must have field definition for field = ${fieldName}`);
+      const isInputValid = fieldDefinition.validateInputValueIfNotNull(fieldValue);
+      if (!isInputValid) {
+        return result(new EntityInvalidFieldValueError(this.entityClass, fieldName, fieldValue));
```

Comment:
May want to consider using AggregateError (https://www.npmjs.com/package/aggregate-error, also will be built into the next version of Node) if we want to return one error for multiple fields. But we'd need to make our error-handling code know how to figure out things like transient vs. permanent AggregateErrors.